R/youth_allowance.R

```diff
@@ -0,0 +1,198 @@
+                     0,
+                     max_rate_March_2018 - ya_income_reduction(ordinary_income, 
+                                                          max_income,
+                                                          is_student,
+                                                          FT_YA_student_lower,
+                                                          FT_YA_student_upper,
+                                                          FT_YA_jobseeker_lower,
+                                                          FT_YA_jobseeker_upper,
+                                                          taper1,
+                                                          taper2))
+                     
+                   
+  payment
+  
+}
+
+ya_parental_income_reduction <- function() {}
+
+ya_partner_reduction <- function(ordinary_income,
+                                 partner_income = 0,
+                                 income_free_area,
+                                 partner_benefit,
+                                 partner_pension,
+                                 partner_dependent_ya) {
+  #social security benifit?
+  
+  
+}
+
+
+ya_income_reduction <- function(ordinary_income,
+                             max_income,
+                             is_student,
+                             max_rate_March_2018,
+                             taper1 = 0.5,
+                             taper2 = 0.6,
+                             FT_YA_student_lower = 437,
+                             FT_YA_student_upper = 524,
+                             FT_YA_jobseeker_lower = 143,
+                             FT_YA_jobseeker_upper = 250) {
+  if_else(is_student,
+          #student
+          if_else(ordinary_income<FT_YA_student_lower,
+                  0,
+                  if_else(ordinary_income < FT_YA_student_upper,
+                          taper1 * (ordinary_income - FT_YA_student_lower),
+                          if_else(ordinary_income < max_income,
+                                  taper1 * (ordinary_income - FT_YA_student_lower) +
+                                    (taper2 - taper1) * (ordinary_income - FT_YA_student_upper),
+                                  #reduce by entire income i.e. dont get YA
+                                  max_rate_March_2018))),
+          #jobseeker
+          if_else(ordinary_income<FT_YA_jobseeker_lower,
+                  0,
+                  if_else(ordinary_income < FT_YA_jobseeker_upper,
+                          taper1 * (ordinary_income - FT_YA_jobseeker_lower),
+                          if_else(ordinary_income < max_income,
+                                  taper1 * (ordinary_income - FT_YA_jobseeker_lower) +
+                                    (taper2 - taper1) * (ordinary_income - FT_YA_jobseeker_upper),
+                                  #reduce by entire income i.e. dont get YA
+                                  max_rate_March_2018))))
+}
```

Comment:
Nit pick: Please ensure all files end with a blank line. Some operating systems / programs need a blank line at the end of text files.